app/src/main/java/com/pmatuki/wowpapers/view/list/WallpaperListActivity.kt

```diff
@@ -31,36 +36,44 @@ class WallpaperListActivity : AppCompatActivity(), WallpaperItemClickListener {
     }
 
     private fun bindViewModel() {
-        viewModel.state.observe(this, { state ->
-            when (state) {
-                WallpaperListState.Loading -> {
-                    binding.apply {
-                        layoutViewLoading.visibility = View.VISIBLE
-                        layoutViewEmpty.visibility = View.GONE
-                    }
+        lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.STARTED) {
```

Comment:
I'm not very familiar with the way to consume Hot Flows, but AFAIK this is the correct approach to correctly subscribe and clean them. 😊

Just out of curiosity, a lot of Hot Flows can be replaced to Cold Flows, making it easier to handle the subscription and cleaning them. I prefer using the cold ones in my project and you can take a look at it [here](https://github.com/igorescodro/alkaa/blob/679ae84c63a57c5cdacd422499bd43c9c486feaf/features/task/src/main/java/com/escodro/task/presentation/list/TaskListViewModel.kt#L26). There is no need to update once your approach is following Google guidelines, but keep in mind that Cold Flows are easy to maintain in the future.